client/src/interface.rs

```diff
@@ -409,6 +409,37 @@ impl Stronghold {
         Ok(res)
     }
 
+    /// [`Stronghold::write_all_to_snapshot`] using a key for encryption that is stored in at the specified client and
+    /// location.
+    pub async fn write_snapshot_with_stored_key(
+        &mut self,
+        key: Location,
+        client_path: Vec<u8>,
+        filename: Option<String>,
+        path: Option<PathBuf>,
+    ) -> StrongholdResult<Result<(), WriteWithKeyError>> {
+        let client_id = ClientId::load_from_path(&client_path, &client_path);
+
+        let clients: Vec<(ClientId, Addr<SecureClient>)> = self.registry.send(GetAllClients).await?;
+
+        let snapshot = self.registry.send(GetSnapshot {}).await?;
+
+        for (id, client) in clients {
+            let data = client.send(GetData {}).await?;
```

Comment:
This part of exporting client's data is crucial, as it contains the key location. At this explicit part I would suggest to extract  the key from client's state, write the snapshot without the decryption key inside it, and reinsert it again.